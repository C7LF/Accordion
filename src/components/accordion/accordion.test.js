import React from 'react'
import renderer from 'react-test-renderer';

import { shallow, mount } from 'enzyme';

import Accordion from './accordion'

describe('Accordion component', () => {

    const mockData = {
        number: '1',
        question: 'question text',
        answer: 'answer text'
    }

    test('Snapshot for accordion component', () => {
        const accordion = renderer.create(<Accordion {...mockData} />)
        let conv = accordion.toJSON()
        expect(conv).toMatchSnapshot()
    })

    test('Should render component at least once', () => {
        const wrapper = shallow(<Accordion {...mockData} />)
        expect(wrapper.find('.accordion__item').length).toBe(1);
    })

    test('Should not render accordion__item if no props are passed', () => {
        const wrapper = shallow(<Accordion />)
        expect(wrapper.find('.accordion__item').length).toBe(0);
    })

    test('Should not render component if order props are only partly passed', () => {
        const wrapper = shallow(<Accordion {...mockData} number='' />)
        expect(wrapper.find('.accordion__item').length).toBe(0);
    })

    test('Should render question div', () => {
        const wrapper = shallow(<Accordion {...mockData} />)
        expect(wrapper.find('.accordion__question').length).toBe(1);
    })

    test('Should render answer div', () => {
        const wrapper = shallow(<Accordion {...mockData} />)
        expect(wrapper.find('.accordion__answer').length).toBe(1);
    })

    test('active class should be present when click is simulated', () => {
        const wrapper = mount(<Accordion {...mockData} />)
        const question = wrapper.find('.accordion__question')
        console.log(wrapper)
        expect(question.length).toBe(1)
        question.simulate('click')
        expect(wrapper.find('.accordion__question').hasClass('active')).toBe(true)
    })
})