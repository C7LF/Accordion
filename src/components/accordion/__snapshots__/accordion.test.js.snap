// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component Snapshot for accordion component 1`] = `
<div
  className="accordion__item"
>
  <div
    className="accordion__question "
    onClick={[Function]}
  >
    <div
      className="question-num"
    >
      <span>
        Q
        1
      </span>
    </div>
    <div
      className="question-text"
    >
      <span>
        question text
      </span>
    </div>
    <div
      className="question-toggle"
    >
      <div
        className="question-toggle__svg"
      >
        <svg>
          plus-icon.svg
        </svg>
      </div>
    </div>
  </div>
  <div
    className="accordion__answer"
    style={
      Object {
        "maxHeight": "0px",
      }
    }
  >
    <p>
      answer text
    </p>
  </div>
</div>
`;
